lua/compe/config.lua

```diff
@@ -35,50 +35,50 @@ Config._normalize = function(config)
   return config
 end
 
-Config.setup = function(config)
-  Config._config = Config._normalize(config)
-end
-
-Config.setup_buffer = function(config)
-  local buf_config = Config._normalize(vim.deepcopy(Config._config))
+Config.setup = function(config, bufnr)
+  if bufnr == nil then
+    Config._config = Config._normalize(config)
+  else
+    local buf_config = Config._normalize(vim.deepcopy(Config._config))
```

Comment:
I think it should be more simply.

```lua
config = Config._normalize(config)

if bufnr == nil then
  Config._config = config
else
  for key, value in pairs(Config._config) do
    if config[key] == nil then
      config[key] = value
    end
  end
  Config._bufnrs[bufnr] = config -- or vim.api.nvim_buf_set_var(bufnr, 'compe_config', config)
end

```

Reply:
I think `vim.api.nvim_buf_set_var` is not required for it... maybe.

Reply:
Probably, the merging logic should be moved to `Config.get`, I think.

Reply:
Much nicer. :+1:

Reply:
> Probably, the merging logic should be moved to `Config.get`, I think.

Doing it in `Config.setup` is a minor optimization, it will be evaluated only once. But I don't know how often `Config.get` is called, so maybe it doesn't really matter.

Reply:
I think we don't worry to performance for it.
If this worse to performance, we can use `utils.cache` module (with versioning).